### DISCUSSION TITLE
Update New API in RpclibserverBase.cpp

### DISCUSSION THREAD

**Discussion by hhhhhhhhmh** [microsoft/AirSim#4100] [Support Q&A] [answered]
### What's your question?

Dear all,
I have put my new function in worldsimapi.cpp and worldsimapibase
then i want to update my api in rpc server.
but when I bind my new api in Rpclibserverbase.cpp
and build it, It will notice me there is an another edition Rpclibserverbase.cpp in UE4Editor-Airsim-Win64-DebugGame.DLL
So that getWorldSimApi() will point to the wrong function.
but in WorldsimApi.cpp my Function have been built succeeded.

**Comment by hhhhhhhhmh** [comment]
Visual Studio told me the source file in the dll is \AirSim\AirLib\src\api\RpcLibServerBase.cpp
so I change that file also. and rebuild it. It seems not work

**Comment by rajat2004** [comment]
I'm not exactly sure what .dll file you;re referring to but it likely gets compiled and is not an actual source file. If you haven't already, then do check out this page - https://microsoft.github.io/AirSim/adding_new_apis/
Can you share details of how it was verified whether adding the API worked or not? And if possible will be great if you could post your change in a branch or somewhere else, that'll make debugging things much easier

**Comment by hhhhhhhhmh** [comment]
> I'm not exactly sure what .dll file you;re referring to but it likely gets compiled and is not an actual source file. If you haven't already, then do check out this page - https://microsoft.github.io/AirSim/adding_new_apis/ Can you share details of how it was verified whether adding the API worked or not? And if possible will be great if you could post your change in a branch or somewhere else, that'll make debugging things much easier

Now, I have already put AirSim Plugin in my own project. And I need to put some new API in my project. So that I build the Function in WorldSimApi.cpp  and WorldSimApiBase.hpp. These places seem to be working fine. But When I want to bind my new API in Rpclibserverbase.cpp. And build it . I can not update this file in UE4Editor-Airsim-Win64-DebugGame.DLL. This .dll file told me it has a source file ,another copy of Rpclibserverbase.cpp in \AirSim\AirLib\src\api\RpcLibServerBase.cpp. But when I change this file in this place. And rebuild it . It seems not work. Visual Studio still told me there is another copy in \AirSim\AirLib\src\api\RpcLibServerBase.cpp

**Comment by jonyMarino** [comment, marked as answer]
@hhhhhhhhmh
if you change the API inside an Unreal Project (like blocks), follow the following path:
1_ do an update_to_git
2_ clean and rebuild AirSim
3_ in the UE project again, do an update_from_git()

**Comment by hhhhhhhhmh** [comment]
> @hhhhhhhhmh if you change the API inside an Unreal Project (like blocks), follow the following path: 1_ do an update_to_git 2_ clean and rebuild AirSim 3_ in the UE project again, do an update_from_git()

Thank you it seems to be working!
